src/lfo.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ring_buffer::RingBuffer;
+use std::f32::consts::PI;
+
+pub struct LFO {
+    wave_table: RingBuffer<f32>,
+    sample_rate_hz: f32,
+    phase_index: f32,
+    freq_hz: f32,
+    amplitude: f32,
+}
+
+impl LFO {
+    pub fn new(sample_rate_hz: f32, size: usize) -> Self {
+        // size determine the resolution of the wave table
+        let mut wave_table = RingBuffer::<f32>::new(size);
+        for i in 0..size {
+            let phase = (i as f32 / size as f32) * 2.0 * PI;
+            wave_table.push(phase.sin());
+        }
+        wave_table.push(0.0);
+        Self {
+            wave_table,
+            sample_rate_hz,
+            freq_hz: 0.0,
+            amplitude: 1.0,
+            phase_index: 0.0,
+        }
+    }
+    /// Set the frequency of the LFO in Hz.
+    /// Example usage
+    /// ```
+    /// let mut lfo = LFO::new(44100.0, 1024);
+    /// lfo.set_frequency(1.0);
+    /// assert_eq!(lfo.freq_hz, 1.0);
+    /// ```
+    pub fn set_frequency(&mut self, freq_hz: f32) {
+        self.freq_hz = freq_hz;
```

Comment:
Here after a new frequency is set, we should also update the values inside the wavetable, otherwise the values inside the wavetable is still the initial values.

Reply:
Same as amplitude and phase